src/io_threads.c

```diff
@@ -285,6 +285,7 @@ static void shutdownIOThread(int id) {
     if (tid == pthread_self()) return;
     if (tid == 0) return;
 
+    pthread_mutex_unlock(&io_threads_mutex[id]);
```

Comment:
I think this is not good to apply this on EVERY io-thread. the problem IIUC is only related to the io-threads which are not active ATM and 
```suggestion
    if (id >= server.active_io_threads_num)
        pthread_mutex_unlock(&io_threads_mutex[id]);
```

Reply:
good catch, fixed.